### PR TITLE
Unifying the VPCI partition mode and sharing mode code (patch series #6)

### PR DESCRIPTION
**Pull request by donsheng**
Changes include:
Remove all CONFIG_PARTITION_MODE from dm/vpci code

Some misra c fix:
     Add @pre for functions
     Add const to function parameters
    Add ASSERT in pci_cfgdata_io_read and pci_cfgdata_io_write

Rename vpci files

Merge partition_mode.c and sharing_mode.c's code into vpci.c

Tracked-On: #2534
Signed-off-by: dongshen dongsheng.x.zhang@intel.com

### CODE REVIEW

File: hypervisor/dm/vpci/vpci.c [hunk 1]
```diff
@@ -82,49 +84,83 @@ static inline bool vpci_is_valid_access(uint32_t offset, uint32_t bytes)
 	return (vpci_is_valid_access_byte(bytes) && vpci_is_valid_access_offset(offset, bytes));
 }
 
+/**
+ * @pre vm->vm_id < CONFIG_MAX_VM_NUM
```
Comment (dongyaozu) [inline hypervisor/dm/vpci/vpci.c]: @pre  vm != NULL

File: hypervisor/dm/vpci/partition_mode.c [hunk 2]
```diff
@@ -137,7 +137,7 @@ void partition_mode_vpci_deinit(const struct acrn_vm *vm)
 	}
 }
 
-void partition_mode_cfgread(struct acrn_vpci *vpci, union pci_bdf vbdf,
+void partition_mode_cfgread(const struct acrn_vpci *vpci, union pci_bdf vbdf,
```
Comment (dongyaozu) [inline hypervisor/dm/vpci/partition_mode.c]: @pre vpci != NULL ?

File: hypervisor/dm/vpci/msix.c [hunk 3]
```diff
@@ -279,6 +279,9 @@ static void vmsix_table_rw(struct pci_vdev *vdev, struct mmio_request *mmio, uin
 
 }
 
+/**
```
Comment (dongyaozu) [inline hypervisor/dm/vpci/msix.c]: @pre io_req != NULL

File: hypervisor/dm/vpci/pci_pt.c [hunk 4]
```diff
@@ -116,7 +116,7 @@ int32_t vdev_pt_cfgread(const struct pci_vdev *vdev, uint32_t offset,
 	return ret;
 }
 
-static void vdev_pt_remap_bar(struct pci_vdev *vdev, uint32_t idx,
```
Comment (dongyaozu) [inline hypervisor/dm/vpci/pci_pt.c]: @pre vdev != NULL
@pre vdev->vpci != NULL
@pre vdev->vpci->vm != NULL

File: hypervisor/dm/vpci/vpci.c [hunk 5]
```diff
@@ -138,13 +174,24 @@ void vpci_init(struct acrn_vm *vm)
 		.len = 4U
```
Comment (dongyaozu) [inline hypervisor/dm/vpci/vpci.c]: @pre vm != NULL ?

File: hypervisor/dm/vpci/vpci.c [hunk 6]
```diff
@@ -161,11 +208,25 @@ void vpci_init(struct acrn_vm *vm)
 	}
 }
 
+/**
```
Comment (dongyaozu) [inline hypervisor/dm/vpci/vpci.c]: @pre vm != NULL

Please check other APIs in this patch too

File: hypervisor/dm/vpci/vpci.c [hunk 7]
```diff
@@ -230,3 +230,281 @@ void vpci_cleanup(const struct acrn_vm *vm)
...
+	if (vdev != NULL) {
+		if ((vmsi_cfgread(vdev, offset, bytes, val) != 0)
+			&& (vmsix_cfgread(vdev, offset, bytes, val) != 0)
+			) {
+			/* Not handled by any handlers, passthru to physical device */
+			*val = pci_pdev_read_cfg(vdev->pdev->bdf, offset, bytes);
+		}
+	}
+}
+
```
Comment (dongyaozu) [inline hypervisor/dm/vpci/vpci.c]: @pre ???

File: hypervisor/dm/vpci/vpci.c [hunk 8]
```diff
@@ -230,3 +230,281 @@ void vpci_cleanup(const struct acrn_vm *vm)
...
+static struct pci_vdev *sharing_mode_find_vdev_sos(union pci_bdf pbdf)
+{
+	struct acrn_vm *vm;
+
+	vm = get_sos_vm();
+
+	return pci_find_vdev_by_pbdf(&vm->vpci, pbdf);
+}
+
```
Comment (dongyaozu) [inline hypervisor/dm/vpci/vpci.c]: @pre ???

File: hypervisor/dm/vpci/vpci.c [hunk 9]
```diff
@@ -230,3 +230,281 @@ void vpci_cleanup(const struct acrn_vm *vm)
 		break;
 	}
 }
+
+static inline bool is_hostbridge(const struct pci_vdev *vdev)
```
Comment (dongyaozu) [inline hypervisor/dm/vpci/vpci.c]: Add @pre for all those new APIs

File: hypervisor/dm/vpci/vpci.c (hunk 1 above)
Comment (donsheng) [reply hypervisor/dm/vpci/vpci.c]: OK, will add, but looks misra c tool does not require this, so I do not add the @pre

File: hypervisor/dm/vpci/vpci.c (hunk 5 above)
Comment (donsheng) [reply hypervisor/dm/vpci/vpci.c]: OK, will add, but looks misra c tool does not require this, so I do not add that

File: hypervisor/dm/vpci/vpci.c (hunk 9 above)
Comment (donsheng) [reply hypervisor/dm/vpci/vpci.c]: OK, will add, but looks misra c tool does not require this, so I do not add the @pre

Comment (donsheng) [reply hypervisor/dm/vpci/vpci.c]: Added @pre to the new/modified APIs

File: hypervisor/dm/vpci/vpci.c (hunk 1 above)
Comment (donsheng) [reply hypervisor/dm/vpci/vpci.c]: Added @pre to the code

**Comment by donsheng** [comment]
Just added @pre to functions per review comments

File: hypervisor/dm/vpci/vpci.c (hunk 8 above)
Comment (donsheng) [reply hypervisor/dm/vpci/vpci.c]: ditto

**Comment by donsheng** [comment]
Looks CI build machine issue?
~/workspace/acrn-hypervisor-premerge-test-mrb-LaaG-image
/tmp/jenkins6411517432102942321.sh: line 27: kill: mrb_process2: arguments must be process or job IDs
/tmp/jenkins6411517432102942321.sh: line 28: kill: mrb_process3: arguments must be process or job IDs
[ ERROR ] Error in file '/storage/cwpqa-acrn-robot/acrn': Importing test library 'device.generator.DeviceGenerator' failed: IndentationError: unexpected indent (bench.py, line 1355)
Traceback (most recent call last):
  File "/storage/cwpqa-acrn-robot/library/device/__init__.py", line 28, in <module>
    from controller.usb_relay import UsbRelay
  File "/storage/cwpqa-acrn-robot/library/controller/usb_relay.py", line 13, in <module>

**Comment by acrnsi** [comment]
OK to verify

**Comment by acrnsi** [comment]
OK to verify

**Pull request merged**